elk/utils_evaluation/utils_evaluation.py

```diff
@@ -58,14 +61,47 @@ def organize(hidden_states, mode):
     raise NotImplementedError("This mode is not supported.")
 
 
-def normalize(data, scale=True, demean=True):
-    # demean the array and rescale each data point
-    data = data - np.mean(data, axis=0) if demean else data
-    if not scale:
-        return data
-    norm = np.linalg.norm(data, axis=1)
-    avgnorm = np.mean(norm)
-    return data / avgnorm * np.sqrt(data.shape[1])
+def normalize(
+    hidden_states,
+    permutation,
+    scale: Literal["original", "none", "elementwise", "projection"] = "original",
+    demean=True,
+):
+    if demean:
+        means = [
+            np.mean(data[permutation[TRAIN_SPLIT_IDX]], axis=0)
+            for data, label in hidden_states
+        ]
+
+        hidden_states = [
+            (data - mean, label) for (data, label), mean in zip(hidden_states, means)
+        ]
+
+    if scale == "original" or scale == "elementwise":
+        if scale == "original":
+            scale_factors = [
+                np.mean(np.linalg.norm(data[permutation[TRAIN_SPLIT_IDX]], axis=1))
+                / np.sqrt(data.shape[1])
+                for data, label in hidden_states
+            ]
+        else:
+            scale_factors = [
+                np.std(data[permutation[TRAIN_SPLIT_IDX]], axis=0)
+                for data, label in hidden_states
+            ]
+        hidden_states = [
+            (data * scale_factor, label)
+            for (data, label), scale_factor in zip(hidden_states, scale_factors)
+        ]
+    elif scale == "projection":
```

Comment:
Maybe we should find a better name for "projection", aren't all the other normalization here not also projections?

Reply:
I've renamed it to layernorm since this is the layernorm projection. Do you prefer this name?